paymentsheet/src/main/java/com/stripe/android/paymentsheet/AddPaymentMethodsAdapter.kt

```diff
@@ -74,42 +75,122 @@ internal class AddPaymentMethodsAdapter(
+            composeView.disposeComposition()
         }
+    }
+
+    internal companion object {
+        private fun calculateViewWidth(parent: ViewGroup, numberOfPaymentMethods: Int): Dp {
+            val targetWidth = parent.measuredWidth - parent.paddingStart - parent.paddingEnd
+            val screenDensity = parent.context.resources.displayMetrics.density
+            val minItemWidth = 100 * screenDensity + 6.0f + 6.0f
+
+            // if all items fit at min width, then span them across the sheet evenly filling it.
+            // otherwise the number of items visible should be a multiple of .5
+            val viewWidth =
+                if (minItemWidth * numberOfPaymentMethods < targetWidth) {
+                    targetWidth / numberOfPaymentMethods
+                } else {
+                    // numVisibleItems is incremented in steps of 0.5 items
+                    // (1, 1.5, 2, 2.5, 3, ...)
+                    val numVisibleItems = (targetWidth * 2 / minItemWidth).toInt() / 2f
+                    targetWidth / numVisibleItems
+                }
 
-        fun bind(paymentMethod: SupportedPaymentMethod) {
-            binding.icon.setImageResource(paymentMethod.iconResource)
-            binding.title.text = itemView.resources.getString(paymentMethod.displayNameResource)
+            return (viewWidth.toInt() / screenDensity).dp
+        }
+    }
+}
+
+@Composable
+fun PaymentMethodUI(
+    viewWidth: Dp,
+    iconRes: Int,
+    title: String,
+    isSelected: Boolean,
+    isEnabled: Boolean,
+    onItemSelectedListener: () -> Unit
+) {
+    val strokeColor = colorResource(
+        if (isSelected) {
+            R.color.stripe_paymentsheet_add_pm_card_selected_stroke
+        } else {
+            R.color.stripe_paymentsheet_add_pm_card_unselected_stroke
+        }
+    )
+
+    val cardBackgroundColor = if (isEnabled) {
+        MaterialTheme.colors.surface
+    } else {
+        colorResource(R.color.stripe_paymentsheet_elements_background_disabled)
+    }
+
+    Card(
+        border = BorderStroke(if (isSelected) 2.dp else 1.5.dp, strokeColor),
+        shape = RoundedCornerShape(6.dp),
+        elevation = if (isSelected) 1.5.dp else 0.dp,
+        backgroundColor = cardBackgroundColor,
+        modifier = Modifier
+            .height(60.dp)
+            .width(viewWidth)
+            .padding(start = 6.dp, end = 6.dp)
+            .selectable(
+                selected = isSelected,
+                enabled = isEnabled,
+                onClick = {
+                    onItemSelectedListener()
+                }
+            )
+    ) {
+        Column {
+            Image(
+                painter = painterResource(iconRes),
+                contentDescription = "",
+                modifier = Modifier
+                    .padding(top = 12.dp, start = 12.dp)
+            )
+            val color = colorResource(R.color.stripe_paymentsheet_title_text)
+            Text(
```

Comment:
nit: Making this a composable named LpmSelectorText might improve readability a little.

Reply:
done